### ISSUE TITLE
Upgrade to Spring-Boot v2.6.x

### ISSUE THREAD

**Issue by ST-DDT** [yidongnan/grpc-spring-boot-starter#601]

**Comment by mschielmann** [comment]
Do you have any details on what needs to be done here? 
Would any help be needed with this?

**Comment by mschielmann** [comment]
Do you have any details on what needs to be done here? 
Would any help be needed with this?

**Comment by ST-DDT** [comment]
Its just an dependency update, so it should be fairly easy.

**Comment by mschielmann** [comment]
Ok, I'll maybe look into this and open an issue / create MR.

I've upgraded my project to SpringBoot 2.6.x and because of the change in the way circular dependencies are handled, my project stopped to boot up:
https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.6-Release-Notes
I needed to add `spring.main.allow-circular-references: true` for it to work as before.

I've trying migrating from `@GrpcClient` to `@GrpcClientBeans`, but it did not work either (beans did not register) so I thought that this issue could help and solve it.

Anyway, I'll test a bit more and either create an issue, with simple example, or an MR.

**Comment by ST-DDT** [comment]
I haven't seen any issues related to circular been references yet. Do you have a stacktrace or a list of beans related to it?

**Comment by mschielmann** [comment]
@ST-DDT I'll try to prepare a minimal reproducible scenario and get back to you. Basically I have all my `@GrpcClients` defined in `@Configuration` class called `GrpcClients`. I am referencing them from a service marked with `@GrpcService` and when I do that I see a circular dependency stating that `GrpcClients` is already under construction. There might be something else with my setup that just came up right now - after SpringBoot update, so I'll have to investigate further and let you know. 

Thank you for your answers!

**Comment by mschielmann** [comment]
Hi @ST-DDT,

So I found what the issue was. In general I have been misusing the `@GrpcClient` annotation, and not knowing how to use `@GrpcClientBean` annotation.

The circular dependency is triggered when in my `@Configuration` class I do something like:
```
@Bean
@GrpcClient("test-client")
public SomeStub testService(SomeStub service) {
    return service;
}
```
This is against the documentation, that is why I tried using `@GrpcClientBean` annotation. But here, I did not do it correctly, I used only the annotation without actually creating the Spring bean with the `@Bean` annotation. 

To sum up, all works ok - it was my mistake. Just for reference, this is the repo explaining my approaches:
https://github.com/mschielmann/grpc-client-spring-boot-issue

Sorry for the trouble and thank you for your support!
